server/neptune/workflows/internal/revision_setter/workflow.go

```diff
@@ -0,0 +1,176 @@
+package revisionsetter
+
+import (
+	"context"
+	"time"
+
+	"github.com/docker/docker/pkg/fileutils"
+	"github.com/pkg/errors"
+	key "github.com/runatlantis/atlantis/server/neptune/context"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/activities"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/activities/github"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/activities/terraform"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/config/logger"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/metrics"
+	"go.temporal.io/sdk/temporal"
+	"go.temporal.io/sdk/workflow"
+)
+
+const (
+	GithubRetryCount    = 3
+	StartToCloseTimeout = 10 * time.Second
```

Comment:
Would maybe bump to 30 seconds at least since some of these with pagination could take time.